### PR TITLE
[s2n] update to 1.3.20

### LINKED ISSUE #26660: [s2n] update to 1.3.20
Library name: s2n-tls (port: s2n)

New version number: 1.3.20

Other information that may be useful (release notes, etc...)
https://github.com/aws/s2n-tls/releases/tag/v1.3.20

### PR DESCRIPTION
**Pull request by JonLiu1993**
Fix #26660

Update s2n to the latest version 1.3.20

### CODE REVIEW

File: ports/s2n/fix-cmake-target-path.patch [hunk 1]
```diff
@@ -12,17 +12,17 @@ index 132f414..c7cfb5f 100644
...
 diff --git a/cmake/s2n-config.cmake b/cmake/s2n-config.cmake
-index 09d1c3a..4e76ed6 100644
+index 9370497..e7cbb88 100644
 --- a/cmake/s2n-config.cmake
 +++ b/cmake/s2n-config.cmake
-@@ -8,9 +8,5 @@ endif()
+@@ -8,9 +8,4 @@ endif()
  list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/modules")
- find_dependency(LibCrypto)
+ find_dependency(crypto)
```
Comment (dg0yt) [inline ports/s2n/fix-cmake-target-path.patch:20]: You must consoldate the patches. This line is patched twice... In the end, we use `find_dependency(OpenSSL COMPONENTS Crypto)`

Comment (JonLiu1993) [reply ports/s2n/fix-cmake-target-path.patch:20]: @dg0yt, Thanks for the reminder, I'll rework the patch

**Comment by armanbilge** [comment]
Already up to 1.3.21
https://github.com/aws/s2n-tls/releases/tag/v1.3.21

**Comment by armanbilge** [comment]
https://github.com/aws/s2n-tls/releases/tag/v1.3.23

**Comment by JonLiu1993** [comment]
> https://github.com/aws/s2n-tls/releases/tag/v1.3.23

@armanbilge, Thanks for your reminder, I'll start upgrading this port after I've dealt with the urgent matter at hand.

**Comment by armanbilge** [comment]
Thanks for your work on this!

**Comment by armanbilge** [comment]
https://github.com/aws/s2n-tls/releases/tag/v1.3.24

**Comment by armanbilge** [comment]
https://github.com/aws/s2n-tls/releases/tag/v1.3.25

**Comment by dg0yt** [comment]
There is one release every week...

**Comment by armanbilge** [comment]
https://github.com/aws/s2n-tls/releases/tag/v1.3.26

**Comment by armanbilge** [comment]
https://github.com/aws/s2n-tls/releases/tag/v1.3.27

**Comment by armanbilge** [comment]
https://github.com/aws/s2n-tls/releases/tag/v1.3.28

**Comment by armanbilge** [comment]
https://github.com/aws/s2n-tls/releases/tag/v1.3.29

**Comment by armanbilge** [comment]
https://github.com/aws/s2n-tls/releases/tag/v1.3.31

**Comment by prateek9623** [comment]
@JonLiu1993 this pr is also needed for this issue https://github.com/microsoft/vcpkg/issues/29083, I tried to update s2n in my PR https://github.com/microsoft/vcpkg/pull/29085. But I am facing issues with the patches, I tried to update the patches, but the thing is s2n requires LC which is the AWS version of `LibCrypto`, which is not replaceable with OpenSSL, as OpenSSL doesn't have EVP_KYBER_512 crypto algorithm. I also looked into if we can create a port [aws-lc](https://github.com/awslabs/aws-lc), but it will create a conflict with OpenSSL, as their headers installs are similar. And I don't know how to handle that. Do you have any suggestions?

**Comment by dg0yt** [comment]
#29236 brought 1.3.35.

**Pull request closed without merging**